frontend/packages/portal-frontend/src/data-explorer-2/query-string-parser.ts

```diff
@@ -46,18 +46,18 @@ const makeFeatureParser = (dimensionKey: DimensionKey) => (
     (d) => d.dataset_id === dataset_id
   );
 
-  const entity_type = dataset ? dataset.entity_type : "custom";
+  const slice_type = dataset ? dataset.slice_type : "custom";
```

Comment:
Oh I see now - I had been confused about what the difference was between a `slice_type` and a `dimension_type`, but it definitely makes sense to distinguish them if `slice_types` can also be "custom". Is this "custom" value also what you use when there is no dimension type (like for transient datasets / custom analysis results)?

Reply:
`"custom"` is used like you're saying but we could've just as well used `null` for that. That's not really what `slice_type` is about.

`slice_type` was born when I realized I was using `entity_type` in two completely different ways:
- When it doesn't matter if it's a sample or feature (that's what `dimension_type` is for!)
  - (In the last round of changes I tried to fix all those) 
- When I wanted to keep track of what was being, well, sliced!
  - In other words, focusing on one row or column of a matrix so it can be plotted 

Consider this matrix:

|            |SOX10|TP53 |FOXA1|
|------------|-----|-----|-----|
| ACH-000001 | 0.1 | 0.1 | 0.1 |
| ACH-000002 | 0.2 | 0.2 | 0.2 |
| ACH-000003 | 0.3 | 0.3 | 0.3 |

`sample_type: "depmap_model"`
`feature_type: "gene"`

I could either slice it like this:
|            |SOX10|
|------------|-----|
| ACH-000001 | 0.1 |
| ACH-000002 | 0.2 |
| ACH-000003 | 0.3 |

or like this:
|            |SOX10|TP53 |FOXA1|
|------------|-----|-----|-----|
| ACH-000001 | 0.1 | 0.1 | 0.1 |

In the first case, my `slice_type` is `"gene"` and the columns form my possible `slice_labels`:
|SOX10|TP53 |FOXA1|
|-----|-----|-----|

In the second case, my `slice_type` is `"depmap_model"` and the rows form my possible `slice_labels`:
|            |
|------------|
| ACH-000001 |
| ACH-000002 |
| ACH-000003 |

It's basically an alternative to keeping track of `is_transpose` all over the place. It can be inferred from the `index_type` and `dataset_id` but it makes the code (at least the UI code) clearer when there's a name for it.

It should always be the case that `sample_type`/`feature_type` match exactly with `index_type`/`slice_type` or otherwise the two are swapped around.

Reply:
Another way to think about this is to consider what you get back from the `/datasets_by_index_type` endpoint:
```js
{
  "depmap_model": [
    {
      "label": "CRISPR (Project Score, Chronos)",
      "dataset_id": "Chronos_Score",
      "slice_type": "gene",
      "index_type": "depmap_model"
    },
    // ...
  ],

  "gene": [
    {
      "label": "CRISPR (Project Score, Chronos)",
      "dataset_id": "Chronos_Score",
      "slice_type": "depmap_model",
      "index_type": "gene"
    },
    // ...
  ]
}
```
The same dataset is represented twice! This is because you can plot it in two different ways. In the UI, if you set your points to "models" it shows you the first list. If you set it to the "genes" it shows you the second list. Then when you pick something you want to plot, it necessarily has to be along the opposite axis. Again, that could be inferred but it's more convenient to just look it up.

Reply:
Ah, I see now! Thank you for the explanation!